### ISSUE TITLE
Some fields cause Jelly_Model::save() to run a query even with no changes

### ISSUE THREAD

**Issue by parshap** [jonathangeiger/kohana-jelly#174]
I've noticed that some fields are causing a database query to be run when a model is saved even when no data has changed. One example is when using a serialized field. Assuming a simple model:

```
class Model_Test extends Jelly_Model
{
    $meta->fields(array(
        'id' => new Field_Primary(),
        'data' => new Field_Serialized(),
    ));
}
```

Calling `->save()` on this model, even with no changed data, will cause a database query to be run. The following causes an INSERT and two UPDATEs:

```
$test = Jelly::factory('test');
$test->data = 'foo';
$test->save();
$test->save();
$test->save();
```

After some investigation, I have identified the problem. In Jelly_Model's save method we iterate over all of the model's values to see if it differs with what is in `$_original`. If it does, we add it to the `$values` array for saving into the database.

```
if ($field->in_db)
{
    // See if field wants to alter the value on save()
    $value = $field->save($this, $value, (bool) $key);

    if ($value !== $this->_original[$column])
    {
        // Value has changed (or has been changed by field:save())
        $values[$field->name] = $value;
```

This is where the problem is. Jelly can represent a single value in two different formats: the way the model represents it, and the way Jelly stores it in the database. For example, with the case of a timestamp field, the model represents the value with an integer, but Jelly stores it in the database depending on the `$format` property (usually a string, as returned by the field's save() method). Except for the case of building the query, Jelly pretty much only uses the model's representation of values.

When the following line is executed, `$value` changes from the model's representation of the value to the database's representation of the value.

```
$value = $field->save($this, $value, (bool) $key);
```

For example, consider the scenario shown above with Test_Model. After the first save `$_original` contains "foo" for the data field, but after the above line `$value` (which used to be "foo") becomes something like 's:3:"foo"'. The comparison following this line will obviously fail and Jelly will think that the data field has changed, even though it hasn't. The same thing happens with a timestamp field when the user sets a time with an integer (i.e., `$model->time = $_SERVER['REQUEST_TIME']`) but then the timestamp field's save() will turn it into a string (even with a `$format` property of just 'U').

To fix this, we must keep the value of the field as the model's representation except for in the `$values` array that is passed to the query builder:

```
if ($field->in_db)
{
    // See if the field's value will be changed after saving
    if ($field->set($field->save($this, $value, (bool) $key)) !== $this->_original[$column])
    {
        // Value has changed (or has been changed by field:save())
        $values[$field->name] = $field->save($this, $value, (bool) $key);
```

We call set() to undo what save() has done and keep the value in the format we expect.

I am not sure if there are any unanticipated side effects to this fix, but none of the unit tests break.

**Comment by parshap** [comment]
There is a commit for this fix here: parshap/kohana-jelly@3c9928663f8575dc86c0bc88eb22fb5f14cf8433

**Comment by parshap** [comment]
Oops. I accidentally clicked "Comment and Close" and closed this ticket. Please reopen.
